wrappers/src/shared_realm_cs.cpp

```diff
@@ -754,6 +754,19 @@ REALM_EXPORT bool shared_realm_remove_type(const SharedRealm& realm, uint16_t* t
     });
 }
 
+REALM_EXPORT bool shared_realm_remove_all(const SharedRealm& realm, NativeException::Marshallable& ex)
+{
+    return handle_errors(ex, [&]() {
+        realm->verify_in_write();
```

Comment:
not sure if I should add more tests, there's `RemoveAllObjectsShouldClearTheDatabase` in theory the only thing that's not being tackled here is making sure it errors when not in write... but I do not see this tested in other functions that are expected to do.

Reply:
thank you, I was unsure how to check on this. How does one get schema from core rather than OS? Core API does not seem clear in this differentiation.

Reply:
I found `get_full_schema` in core.... but it is a private method

Reply:
I forgot that actually we have a long-standing ticket somewhere where we request that the schema at object store level should be the same that core sees. 
I am also not sure of how to do it to be honest 😁 I would need to take a look to do it. 
Another thing, we need to be sure that this works fine with Flexible Sync. So, if for example we call this method, this should only delete the objects that have been synced, and doesn't delete more on the server

Reply:
`ObjectStore::schema_from_group(realm->read_group())` is probably an overkill for this as you don't really care about the entire schema - you only want the tables (hint: you can get those from the group).